src/platform/nrfconnect/shell/ChipShell.cpp

```diff
@@ -0,0 +1,212 @@
+    }
+    else
+    {
+        shell_print(shell, "Getting Setup Pin Code failed with error: %d", error);
+        result = -ENOEXEC;
+    }
+
+    return result;
+}
+
+static int ChipShellProcessDiscriminator(const struct shell * shell, size_t argc, char ** argv)
+{
+    int result       = 0;
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    uint16_t discriminator;
+
+    error = ConfigurationMgr().GetSetupDiscriminator(discriminator);
+    if (error == CHIP_NO_ERROR)
+    {
+        shell_print(shell, "%d", discriminator);
+    }
+    else
+    {
+        shell_print(shell, "Getting Setup Discriminator failed with error: %d", error);
+        result = -ENOEXEC;
+    }
+
+    return result;
+}
+
+static int ChipShellProcessVendorId(const struct shell * shell, size_t argc, char ** argv)
+{
+    int result       = 0;
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    uint16_t vendorId;
+
+    error = ConfigurationMgr().GetVendorId(vendorId);
+    if (error == CHIP_NO_ERROR)
+    {
+        shell_print(shell, "%d", vendorId);
+    }
+    else
+    {
+        shell_print(shell, "Getting Vendor ID failed with error: %d", error);
+        result = -ENOEXEC;
+    }
+
+    return result;
+}
+
+static int ChipShellProcessProductId(const struct shell * shell, size_t argc, char ** argv)
+{
+    int result       = 0;
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    uint16_t productId;
+
+    error = ConfigurationMgr().GetProductId(productId);
+    if (error == CHIP_NO_ERROR)
+    {
+        shell_print(shell, "%d", productId);
+    }
+    else
+    {
+        shell_print(shell, "Getting Product ID failed with error: %d", error);
+        result = -ENOEXEC;
+    }
+
+    return result;
+}
+
+static int ChipShellProcessManualPairingCode(const struct shell * shell, size_t argc, char ** argv)
+{
+    int result       = 0;
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    std::string manualPairingCode;
+
+    error = GetManualPairingCode(manualPairingCode, chip::RendezvousInformationFlags(chip::RendezvousInformationFlag::kBLE));
+    if (error == CHIP_NO_ERROR)
+    {
+        shell_print(shell, "%s", manualPairingCode.c_str());
+    }
+    else
+    {
+        shell_print(shell, "Getting Manual Pairing Code failed with error: %d", error);
+        result = -ENOEXEC;
+    }
+
+    return result;
+}
+
+static int ChipShellProcessBleAdvertising(const struct shell * shell, size_t argc, char ** argv)
+{
+    if (ConnectivityMgr().IsBLEAdvertisingEnabled())
+    {
+        shell_print(shell, "1");
+    }
+    else
+    {
+        shell_print(shell, "0");
+    }
+
+    return 0;
+}
+
+static int ChipShellProcessNfcEmulation(const struct shell * shell, size_t argc, char ** argv)
+{
+    if (NFCMgr().IsTagEmulationStarted())
+    {
+        shell_print(shell, "1");
+    }
+    else
+    {
+        shell_print(shell, "0");
+    }
+
+    return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(chipCommands, SHELL_CMD_ARG(qrcode, NULL, "qrcode", ChipShellProcessQRCode, 0, 0),
```

Comment:
nit: I would prefer to use naming such as `cmd_qr_code` as it is done almost everywhere in Zephyr (also in CHIP shell), plus static inline function with prefix `ChipShellProcess` seems strange for me.